src/lib.rs

```diff
@@ -35,33 +35,40 @@ use colored::Colorize;
 use std::borrow::Cow;
 use std::io::Write;
 use std::sync::{atomic::AtomicBool, Arc};
+use std::thread::sleep;
 use std::thread::{self, JoinHandle};
 use std::time::Duration;
-use std::thread::sleep;
 
-mod utils;
-mod spinner_data;
-mod spinner_enum;
+pub mod spinners;
 mod streams;
+mod utils;
 
+use spinners::SpinnerFrames;
+pub use streams::Streams;
 pub use utils::Color;
 use utils::{colorize, delete_last_line};
-use spinner_data::SPINNER_FRAMES;
-pub use spinner_enum::Spinners;
-pub use streams::Streams;
 
 /// Terminal spinner.
-#[derive(Debug)]
 pub struct Spinner {
     thread_handle: Option<JoinHandle<()>>,
     /// This struct has an `Arc<AtomicBool>` field, which is later used in the `stop` type methods to stop the thread printing the spinner.
     still_spinning: Arc<AtomicBool>,
-    spinner_type: Spinners,
+    spinner_frames: SpinnerFrames,
     msg: Cow<'static, str>,
     stream: Streams,
     color: Option<Color>,
 }
 
+#[macro_export]
+macro_rules! spinner {
```

Comment:
Why are we defining a macro that is never used here? My best guess is that it's for creating a new spinner by the user, but we should at least make a new example file showing the user how to use it.

Reply:
Added documentation for custom spinners using the `spinner!` macro in 41e21ed8f1f51803cb9325d8718b3d6bfeedc7ee